updatesnap/updatesnapyaml.py

```diff
@@ -40,10 +40,9 @@ def get_working_branch(self, project_url):
                 break
         return working_branch
 
-    def get_yaml_file(self, project_url):
+    def get_yaml_file(self, project_url, yaml_path):
         """ Searches in a project for the 'snapcraft.yaml' file and
```

Comment:
I dislike how this has been implemented. This presumes that `snapcraft.yaml` will be passed by default in the `yaml_path` parameter, and if another path is passed, it still will try to search in `snap/snapcraft.yaml`. I think that the default value in `args` should be `None`, and in this function, if the `yaml_path` value is `None`, should search in the common places using the old code, and if it isn't `None`, should search only in the passed path and fail if it is an invalid file or it doesn't exist.